src/codegate/db/connection.py

```diff
@@ -446,6 +475,21 @@ async def get_workspace_by_name(self, name: str) -> Optional[Workspace]:
         )
         return workspaces[0] if workspaces else None
 
+    async def get_workspace_by_name(self, name: str) -> Optional[Workspace]:
```

Comment:
should we rename this to deal with the expectation that we're dealing with soft deleted or archived workspaces?